File: fsharp/bob/Bob.fs
module Bob

let response (input: string): string = 
    let trimmed = input.Trim()
    let trimmedChars = trimmed.ToCharArray()
    let isQuestion = 
        Array.tryLast trimmedChars = Some '?'

    let isLetter (x:char) = System.Char.IsLetter(x)
    let isUpper =
        let matches = trimmed.ToUpper() = trimmed
        let hasLetters = Array.tryFind isLetter trimmedChars <> None
        matches && hasLetters
    let notControl (x:char) = not (System.Char.IsControl(x))

    match trimmed with
    | _ when trimmed.Length = 0 || Array.tryFind notControl trimmedChars = None -> "Fine. Be that way!"
    | _ when isUpper && isQuestion -> "Calm down, I know what I'm doing!"
    | _ when isUpper -> "Whoa, chill out!"
    | _ when isQuestion -> "Sure."
    | _ -> "Whatever."

File: csharp/bob/Bob.cs
using System;

public static class Bob
{
    public static string Response(string statement)
    {
        statement = statement.Trim();
        if (string.IsNullOrWhiteSpace(statement))
            return "Fine. Be that way!";
        else if (statement.ToUpper() == statement && statement.ToLower() != statement && statement.EndsWith('?'))
            return "Calm down, I know what I'm doing!";
        else if (statement.EndsWith('?'))
            return "Sure.";
        else if (statement.ToUpper() == statement && statement.ToLower() != statement)
            return "Whoa, chill out!";
        else
            return "Whatever.";
    }
}